### DISCUSSION TITLE
How can I detect that the app is running in Skia.Gtk?

### DISCUSSION THREAD

**Discussion by beto-rodriguez** [unoplatform/uno#8677] [Q&A]
In my case, I subscribe to the `UIElement.PointerWheelChanged` event to [zoom in/out](https://github.com/beto-rodriguez/LiveCharts2/blob/master/src/skiasharp/LiveChartsCore.SkiaSharpView.Uno/CartesianChart.xaml.cs#L986) a chart.

The problem is that based on the wheel delta I decide if the zoom goes in or goes out *(zoom in if positive, zoom out if negative)*, so far so good, it works in WPF and UWP the same way.

But when my app is running on Skia.Gtk then the wheel delta is inverted *(zoom in negative, zoom out positive)* and this causes the zoom to behave different on this platform.

How can I detect that the app is running in Skia.Gtk?
Is there another alternative for my problem?

**Comment by jeromelaban** [comment]
This looks like a bug! @dr1rrb will likely know more, and we'll fix this.

**Comment by beto-rodriguez** [comment]
Thanks @jeromelaban, also the wheel delta value is different in the 3 platforms, this behavior does not affect my case but I think it would be nice to get the same value in all the platforms.

**Reply by dr1rrb** [reply]
Which 3 platforms? We tried to get as close as possible on all platforms, but on wasm for instance it depends of the browser ...

**Reply by beto-rodriguez** [reply]
UWP, WPF and Skia.Gtk (I ran it on windows)

**Comment by dr1rrb** [comment]
Well that's weird as we are using this `PointerWheelChanged` event for the scrolling on skia (on other platforms like wasm, we let the system handle it by itself). Does the scrolling goes in the right direction when you use the wheel on a `ScrollViewer`? Also, if you tested on windows, did you enabled in the OS settings the "reverse scrolling"?

**Reply by beto-rodriguez** [reply]
> Does the scrolling goes in the right direction when you use the wheel on a ScrollViewer?

It does, but the scrolling is sluggish, it seems that the scroll does not happen every time I move the mouse wheel, it just gets updated in random intervals.

> Also, if you tested on windows, did you enabled in the OS settings the "reverse scrolling"?

Yes I tested on Windows, No reverse scrolling is not enabled, is it required?
